packages/lisk-p2p/src/p2p.ts

```diff
@@ -12,59 +12,185 @@
 		penalty;
-	};
-	// TODO
-	public getNetworkStatus = (): NetworkStatus => true;
-	// TODO
-	public getNodeStatus = (): P2PNodeStatus => true;
-	// TODO
-	public request = async (
-		packet: P2PRequestPacket,
-	): Promise<P2PResponsePacket> => {
-		const response = packet;
-
-		return Promise.resolve(response);
-	};
-
-	public send = (message: P2PMessagePacket): void => {
+	}
+
+	// TODO ASAP: public getNetworkStatus(): NetworkStatus {};
+
+	public async request<T>(
+		packet: P2PRequestPacket<T>,
+	): Promise<P2PResponsePacket> {
+		return Promise.resolve({ data: packet });
+	}
+
+	public send<T>(message: P2PMessagePacket<T>): void {
 		message;
-		// TODO
-	};
-	public setNodeStatus = (nodeStatus: P2PNodeStatus): void => {
-		nodeStatus;
-		// TODO
-	};
-	// TODO
-	public start = async (): Promise<void> => this.peerPool.start();
-	// TODO
-	public stop = async (): Promise<void> => this.peerPool.stop();
+		// TODO ASAP
+	}
+
+	public get nodeStatus(): P2PNodeStatus {
+		return this._nodeStatus;
+	}
+
+	/**
+	 * This is not a declared as a setter because this method will need
+	 * invoke an async RPC on Peers to give them our new node status.
+	 */
+	public applyNodeStatus(value: P2PNodeStatus): void {
+		this._nodeStatus = value;
+		// TODO ASAP: Pass to PeerPool -> connected Peer objects
+	}
+
+	private async _startPeerServer(): Promise<void> {
+		this._scServer.on(
+			'connection',
+			(socket: any): void => {
+				const queryObject = querystring.parse(socket.request.url);
+				if (
+					typeof queryObject.wsPort !== 'string' ||
+					typeof queryObject.os !== 'string' ||
+					typeof queryObject.version !== 'string'
+				) {
+					super.emit(EVENT_FAILED_TO_ADD_INBOUND_PEER);
+				} else {
+					const wsPort: number = parseInt(queryObject.wsPort, BASE_10_RADIX);
+					const peerId = Peer.constructPeerId(socket.remoteAddress, wsPort);
+					const existingPeer = this._peerPool.getPeer(peerId);
+					if (existingPeer === undefined) {
+						const peer = new Peer({
+							inboundSocket: socket,
+							ipAddress: socket.remoteAddress,
+							os: queryObject.os,
+							version: queryObject.version,
+							wsPort,
+							nodeStatus: this._nodeStatus,
+						});
+						this._peerPool.addPeer(peer);
+						super.emit(EVENT_NEW_INBOUND_PEER, peer);
+						super.emit(EVENT_NEW_PEER, peer);
+						this._newPeers.add(peer.peerConfig);
+					} else {
+						existingPeer.inboundSocket = socket;
+						this._newPeers.add(existingPeer.peerConfig);
+					}
+				}
+			},
+		);
+		this._httpServer.listen(this._config.wsPort);
+
+		return new Promise<void>(resolve => {
+			this._scServer.once('ready', () => {
+				resolve();
+			});
+		});
+	}
+
+	private async _stopPeerServer(): Promise<void> {
+		// TODO ASAP: Test this and check for potential failure scenarios.
+		return new Promise<void>(resolve => {
+			this._scServer.close(() => {
+				resolve();
+			});
+		});
+	}
+
+	public async _loadListOfPeerListsFromSeeds(
```

Comment:
a public method name should start without underscore

Reply:
Ah it should be private.